src/opentslm/model/llm/OpenTSLMFlamingo.py

```
@@ -152,6 +157,39 @@ def _infer_decoder_layers_attr_name(model):
+            return super(FlamingoLMMixin, self.model.lang_encoder).forward(
+                inputs_embeds=inputs_embeds,
+                attention_mask=attention_mask,
+                labels=labels,
+            )
+        finally:
+            self.model.lang_encoder.clear_conditioned_layers()
+
+    def _condition_media_locations(self, input_ids: torch.Tensor):
+        media_locations = input_ids == self.model.media_token_id
+        attend_previous = (
+            (random.random() < 0.5)
```

Ruff (0.15.9)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)